include/sdfg/structured_control_flow/structured_loop.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "sdfg/structured_control_flow/control_flow_node.h"
+#include "sdfg/structured_control_flow/sequence.h"
+#include "sdfg/symbolic/symbolic.h"
+
+namespace sdfg {
+
+namespace builder {
+class StructuredSDFGBuilder;
+}
+
+namespace structured_control_flow {
+
+class StructuredLoop : public ControlFlowNode {
+    friend class sdfg::builder::StructuredSDFGBuilder;
+
+   protected:
+    StructuredLoop(size_t element_id, const DebugInfo& debug_info);
+
+   public:
+    virtual ~StructuredLoop() = default;
+
+    StructuredLoop(const StructuredLoop& node) = delete;
+    StructuredLoop& operator=(const StructuredLoop&) = delete;
+
+    virtual const symbolic::Symbol& indvar() const = 0;
```

Comment:
Why shall we require the symbolic description on this type of node? I was thinking of an interface maybe with root as the only virtual func?